### ISSUE TITLE
tflite: output tensor size is not variable.

### ISSUE THREAD

**Issue by wenhui7909** [tensorflow/tensorflow#20052]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Ubuntu 17.10
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: tensorflow 1.8
- **Python version**:  2.7
- **Bazel version (if compiling from source)**: 0.13.0
- **GCC/Compiler version (if compiling from source)**: 6.4
- **CUDA/cuDNN version**: None
- **GPU model and memory**: GeForce GTX 750
- **Exact command to reproduce**: run mtcnn pnet model in tflite android demo of tflitecamerademo.

### Describe the problem
Use mtcnn pnet source code to generate pnet.pb file, 
then convert pnet.pb to pnet.tflite file.
when run pnet.tflite in  tflitecamerademo and input size is not the same as the number of when converting the tflite, the error log is "num_input_elements != num_output_elements (42050 != 48050)".
I guess output tensor need resize because I resize the input tensor.
But I can't find the API of output tensor resize.

### Source code / logs
The pnet code as below:
    (self.feed('data')  #pylint: disable=no-value-for-parameter, no-member
    .conv(3, 3, 10, 1, 1, padding='VALID', relu=False, name='conv1')
     .prelu(name='PReLU1')
     .max_pool(2, 2, 2, 2, name='pool1')
     .conv(3, 3, 16, 1, 1, padding='VALID', relu=False, name='conv2')
     .prelu(name='PReLU2')
     .conv(3, 3, 32, 1, 1, padding='VALID', relu=False, name='conv3')
     .prelu(name='PReLU3')
     .conv(1, 1, 2, 1, 1, relu=False, name='conv4-1')
      .softmax(3,name='prob1'))

    (self.feed('PReLU3') #pylint: disable=no-value-for-parameter
    .conv(1, 1, 4, 1, 1, relu=False, name='conv4-2'))
generate pb file as below:
   data = tf.placeholder(name="input", dtype=tf.float32, shape=(None, None, None, 3))
   pnet = PNet({'data':data})
   pnet.load(os.path.join(model_path, 'det1.npy'), sess)

    frozen_graphdef = tf.graph_util.convert_variables_to_constants(
            sess, sess.graph_def, ['pnet/conv4-2/BiasAdd', 'pnet/prob1'])
   model_f = tf.gfile.GFile("/home/hwh/pnet_none.pb", "wb")

convert to tflite file as below:
./bazel-bin/tensorflow/contrib/lite/toco/toco    --input_file=/home/hwh/pnet.pb   --input_format=TENSORFLOW_GRAPHDEF  --output_format=TFLITE   --output_file=/home/hwh/pnet.tflite --inference_type=FLOAT   --input_type=FLOAT --input_arrays=pnet/input   --output_arrays=pnet/prob1,pnet/conv4-2/BiasAdd --input_shapes=1,320,320,3

then I run this tflite file in tflitecamerademo, and before tflite run, 
I resize input tensor to (1, 299, 299, 3).

when run the error log as below:
    06-15 16:14:36.500 12080-12275/android.example.com.tflitecamerademo E/AndroidRuntime: FATAL EXCEPTION: CameraBackground
    Process: android.example.com.tflitecamerademo, PID: 12080
    java.lang.NullPointerException: Internal error: Can not allocate memory for the given inputs: tensorflow/contrib/lite/kernels/reshape.cc:68 num_input_elements != num_output_elements (42050 != 48050)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(Native Method)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:123)
        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:166)
        at com.example.android.tflitecamerademo.ImageClassifierFloatInception.runInference(ImageClassifierFloatInception.java:121)
        at com.example.android.tflitecamerademo.ImageClassifier.classifyFrame(ImageClassifier.java:117)
        at com.example.android.tflitecamerademo.Camera2BasicFragment.classifyFrame(Camera2BasicFragment.java:697)
        at com.example.android.tflitecamerademo.Camera2BasicFragment.-wrap0(Camera2BasicFragment.java)
        at com.example.android.tflitecamerademo.Camera2BasicFragment$4.run(Camera2BasicFragment.java:592)
        at android.os.Handler.handleCallback(Handler.java:815)
        at android.os.Handler.dispatchMessage(Handler.java:104)
        at android.os.Looper.loop(Looper.java:194)
        at android.os.HandlerThread.run(HandlerThread.java:61)

**Comment by andrehentz** [comment]
It looks like the model you are using has a Reshape op with a fixed size. The easiest workaround is to reconvert using --input_shapes=1,299,299,3. Does that work? Otherwise you'll need to change the resize code in the java app.

**Comment by wenhui7909** [comment]
Yes, when input_shapes = 1, 299, 299, 3 is ok.
but my pb is generated by input tensor of (None, None, None, 3),
I hope the input tensor size is variable.
And I can resize input size in java code,
and my output tensor size is fit for (1, 299, 299, 3) input,
but output is reported error.
I made an experiment that mean op, input tensor size is variable, output tensor is fixed, it is ok.
But now my net output size will change with input tensor.
So I think it is perfect that the output is auto fit for input tensor,
And my java code need allocate an enough memory.
Thanks.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by suharshs** [comment]
For now you will have to fix your shape, this is a known bug that we are tracking and plan to fix.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @andrehentz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by gargn** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @suharshs: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sanjnair** [comment]
Any updates on this issue? Is this fixed?

**Comment by shahid14** [comment]
https://github.com/tensorflow/tensorflow/issues/22377#issuecomment-444550814

**Comment by zjd1988** [comment]
Hi @suharshs

when will you fix this problem?

thanks

**Comment by SamanvayGA** [comment]
Wad it fixed in a newer version?

**Comment by elaith9** [comment]
is this fixed?
